File: 剑指Offer32-II.从上到下打印二叉树II/剑指Offer32-II.从上到下打印二叉树II.cpp
/*
 * @Descroption: 剑指 Offer 32 - II. 从上到下打印二叉树 II
 * @Author: EmoryHuang
 * @Date: 2021-07-03 13:58:31
 * @解题思路:
 * 按照层序遍历的方法打印即可
 */

/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
   public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if (!root) return ans;
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()) {
            int n = q.size();
            vector<int> tmp;
            for (int i = 0; i < n; i++) {
                auto node = q.front();
                q.pop();
                tmp.push_back(node->val);
                if (node->left) q.push(node->left);
                if (node->right) q.push(node->right);
            }
            ans.push_back(tmp);
        }
        return ans;
    }
};

File: Solution/剑指Offer32-II.从上到下打印二叉树II/剑指Offer32-II.从上到下打印二叉树II.py
'''
Descroption: 剑指 Offer 32 - II. 从上到下打印二叉树 II
Author: EmoryHuang
Date: 2021-07-03 13:54:46
解题思路:
按照层序遍历的方法打印即可
'''

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        ans = []
        if not root:
            return ans
        q = deque()
        q.append(root)
        while q:
            n = len(q)
            tmp = []
            for _ in range(n):
                node = q.popleft()
                tmp.append(node.val)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
            ans.append(tmp)
        return ans
